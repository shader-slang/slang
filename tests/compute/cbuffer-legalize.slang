//TEST(compute):COMPARE_COMPUTE:-cpu -shaderobj
//TEST(compute):COMPARE_COMPUTE:-shaderobj

//TEST_INPUT: uniform(data=[1 2 3 4]):name=C.p.c
//TEST_INPUT: Texture2D(size=4, content = one):name=C.p.t
//TEST_INPUT: Sampler:name=C.p.s
//TEST_INPUT: ubuffer(data=[0 0 0 0], stride=4):out,name=outputBuffer

struct P
{
	uint4 			c;
	Texture2D 		t;
	SamplerState 	s;
};

float4 test(P p)
{
	return p.t.SampleLevel(p.s, float2(0.0), 0) + p.c;
}

cbuffer C
{
	P p;
};

RWStructuredBuffer<float> outputBuffer;

[numthreads(1, 1, 1)]
void computeMain(uint3 dispatchThreadID : SV_DispatchThreadID)
{
	float4 outVal = test(p);

    outputBuffer[0] = outVal.x;
	outputBuffer[1] = outVal.y;
	outputBuffer[2] = outVal.z;
	outputBuffer[3] = outVal.w;
}