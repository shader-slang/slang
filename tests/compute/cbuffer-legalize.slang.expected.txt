40000000
40400000
40800000
40A00000
